dask_expr/tests/test_repartition.py

```diff
@@ -34,6 +34,65 @@ def test_repartition_combine_similar(kwargs):
     assert_eq(result, expected_pdf)
 
 
+@pytest.mark.parametrize("type_ctor", [lambda o: o, tuple, list])
+def test_repartition_noop(type_ctor):
+    pdf = lib.DataFrame({"x": [1, 2, 4, 5], "y": [6, 7, 8, 9]}, index=[-1, 0, 2, 7])
+    df = from_pandas(pdf, npartitions=2)
+    ds = df.x
+
+    # DataFrame method
+    df2 = df.repartition(divisions=type_ctor(df.divisions))
+    assert (
+        len(
+            [
+                x
+                for x in df2.expr.optimize(fuse=False).walk()
+                if isinstance(x, Repartition)
+            ]
+        )
+        == 0
+    )
+
+    # Top-level dask.dataframe method
+    df3 = repartition(df, divisions=type_ctor(df.divisions))
+    assert (
+        len(
+            [
+                x
+                for x in df3.expr.optimize(fuse=False).walk()
+                if isinstance(x, Repartition)
+            ]
+        )
+        == 0
+    )
+
+    # Series method
+    ds2 = ds.repartition(divisions=type_ctor(ds.divisions))
+    assert (
+        len(
+            [
+                x
+                for x in ds2.expr.optimize(fuse=False).walk()
+                if isinstance(x, Repartition)
+            ]
+        )
+        == 0
+    )
+
+    # Top-level dask.dataframe method applied to a Series
+    ds3 = repartition(ds, divisions=type_ctor(ds.divisions))
+    assert (
+        len(
+            [
+                x
+                for x in ds3.expr.optimize(fuse=False).walk()
+                if isinstance(x, Repartition)
+            ]
+        )
+        == 0
+    )
```

Comment:
```suggestion
    def assert_not_repartitions(expr, fuse=False):
        repartitions = [
            x for x in expr.optimize(fuse=fuse).walk() if isinstance(x, Repartition)
        ]
        assert len(repartitions) == 0

    # DataFrame method
    df2 = df.repartition(divisions=type_ctor(df.divisions))
    assert_not_repartitions(df2.expr)

    # Top-level dask.dataframe method
    df3 = repartition(df, divisions=type_ctor(df.divisions))
    assert_not_repartitions(df3.expr)

    # Series method
    ds2 = ds.repartition(divisions=type_ctor(ds.divisions))
    assert_not_repartitions(ds2.expr)

    # Top-level dask.dataframe method applied to a Series
    ds3 = repartition(ds, divisions=type_ctor(ds.divisions))
    assert_not_repartitions(ds3.expr)
```

Maybe slightly easier on the eyes?

Reply:
oh yes, nice catch!